dataprofiler/profilers/profile_builder.py

```diff
@@ -1355,11 +1362,118 @@ def _update_row_statistics(self, data, sample_ids=None):
             self.row_has_null_count = len(null_in_row_count)
             self.row_is_null_count = len(null_rows)
 
+    def _update_correlation(self, clean_samples):
+        """
+        Update correlation matrix for cleaned data.
+
+        :param clean_samples: the input cleaned dataset
+        :type clean_samples: dict()
+        """
+        columns = self.options.correlation.columns
+        if columns is None:
+            for col in self._profile:
+                if self._profile[col].profile['data_type'] not in ['int', 'float'] \
+                or self._profile[col].profile['statistics']['null_count'] > 0:
+                    clean_samples.pop(col)
+        if len(clean_samples) <= 1:
+            return
+
+        data = pd.DataFrame(clean_samples)
+        data = data.apply(pd.to_numeric, errors='coerce')
+        correlation_matrix = data.corr()
+
+        if self.correlation_matrix is None:
+            self.correlation_matrix = correlation_matrix
+            return
+
+        # currently return None with the exising correlation
+        # TODO: implement the update with the exising correlation
+        warnings.warn("Currently the update with existing correlation "
+                      "doesnt work.")
```

Comment:
I'll think about this, but I think we should polish this text a little more. Something along the lines of merging correlations will be available in a future update, as a result the correlations have been turned off for this profile.

Reply:
modified the message. Does that look better?

Reply:
I'm not a fan of the phrase doesn't work as oppose to coming in a future release.

Reply:
There also needs to be description of what that means, e.g. how does that affect the user's profile.